OpenRA.Mods.Common/Traits/Render/WithBuildingExplosion.cs

```diff
@@ -19,6 +19,7 @@ namespace OpenRA.Mods.Common.Traits
 	class WithBuildingExplosionInfo : ITraitInfo, Requires<BuildingInfo>
 	{
 		[Desc("Explosion sequence name to use")]
+		/* [SequenceReference] TODO: not on the same actor */
```

Comment:
You should extend the SequenceReferenceAttribute to accept an argument that overrides the sequence name (e.g. `[SequenceReference("explosion")]`)